assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * PostSearcherAutoSuggest component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+	Combobox,
+	ComboboxInput,
+	ComboboxPopover,
+	ComboboxList,
+	ComboboxOption,
+} from '@reach/combobox';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { useDebouncedState } from '../hooks/useDebouncedState';
+
+const PostSearcherAutoSuggest = ( { setCanSubmit, setMatch } ) => {
+	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const debouncedValue = useDebouncedState( searchTerm, 200 );
+	const [ results, setResults ] = useState( [] );
+	const noResultsMessage = __( 'No results found', 'google-site-kit' );
+
+	const postSearch = async ( query ) => {
+		try {
+			const queryResults = await API.get( 'core', 'search', 'post-search',
+				{ query: encodeURIComponent( query ) },
+				{ useCache: false }
+			);
+
+			if ( Array.isArray( queryResults ) ) {
+				setResults( queryResults );
+			} else {
+				setResults( [] );
+			}
```

Comment:
Is it possible for a non-array to come back from a successful request? If so, I would suggest we fix that in the endpoint. It should always return an array and we should be able to trust that it does when the response is successful. If there is some reason where it is still possible then we should leave a comment explaining that here.

If `queryResults` is always an array here then let's simplify it. If we need to keep this conditional, then something is wrong or there is some extra context here which isn't obvious as to how this might happen 🙂 
```suggestion
			setResults( queryResults );
```